lib/spack/spack/cmd/create.py

```diff
@@ -138,39 +135,47 @@ def setup_parser(subparser):
 
 class ConfigureGuesser(object):
     def __call__(self, stage):
-        """Try to guess the type of build system used by the project. Set the
-        appropriate default configure, build, and install instructions."""
-
-        # Default configure instructions
-        configureDict = {
-            'autotools': "configure('--prefix={0}'.format(prefix))",
-            'cmake':     "cmake('.', *std_cmake_args)",
-            'scons':     "",
-            'python':    "",
-            'r':         "",
-            'unknown':   "# FIXME: Unknown build system"
-        }
-
-        # Default build instructions
-        buildDict = {
-            'autotools': "make()",
-            'cmake':     "make()",
-            'scons':     "scons('prefix={0}'.format(prefix))",
-            'python':    "",
-            'r':         "",
-            'unknown':   "make()",
-        }
+        """Try to guess the type of build system used by the project.
+        Set the appropriate default installation instructions and any
+        necessary extensions for Python and R."""
 
-        # Default install instructions
+        # Default installation instructions
         installDict = {
-            'autotools': "make('install')",
-            'cmake':     "make('install')",
-            'scons':     "scons('install')",
-            'python':    "python('setup.py', 'install', " +
-                         "'--prefix={0}'.format(prefix))",
-            'r':         "R('CMD', 'INSTALL', '--library={0}'.format(" +
-                         "self.module.r_lib_dir), self.stage.archive_file)",
-            'unknown':   "make('install')",
+            'autotools': """\
+        # FIXME: Modify the configure line to suit your build system here.
+        configure('--prefix={0}'.format(prefix))
+
+        # FIXME: Add logic to build and install here.
+        make()
+        make('install')""",
+
+            'cmake':     """\
+        with working_dir('spack-build', create=True):
+            # FIXME: Modify the cmake line to suit your build system here.
+            cmake('..', *std_cmake_args)
+
+            # FIXME: Add logic to build and install here.
+            make()
+            make('install')""",
+
+            'scons':     """\
+        # FIXME: Add logic to build and install here.
+        scons('prefix={0}'.format(prefix))
+        scons('install')""",
+
+            'python':    """\
+        # FIXME: Add logic to build and install here.
+        python('setup.py', 'install', '--prefix={0}'.format(prefix))""",
+
+            'R':         """\
+        # FIXME: Add logic to build and install here.
+        R('CMD', 'INSTALL', '--library={0}'.format(self.module.r_lib_dir),""" +
+            " self.stage.archive_file)""",
```

Comment:
I was going to be touching the R template for #1000 but it seems like it is best to do that here. Could you change the R build command above to
```
R('CMD', 'INSTALL', '--library={0}'.format(self.module.r_lib_dir),""" +
 +            " self.stage.source_path)""",
```
Then I will make sure to leave my local change to this file out of #1000.

Thanks.